scripts/snapshot-restore-balance/snapshot-balances.js

```diff
@@ -0,0 +1,81 @@
+const { ApiPromise, WsProvider, Keyring } = require('@polkadot/api');
+const fs = require('fs');
+
+const WS_ENDPOINT = 'wss://testnet-rpc.zkverify.io';
+const DISCARDED_ACCOUNTS = [
+        "5D9txxK9DTvgCznTjJo7q1cxAgmWa83CzHvcz8zhBtLgaLBV",
+        "5ETuZEyLnfVzQCaDM8aQCcsNnz6xjPKvQCtqynCLqwng8QLd",
+        "5D29UEzgStCBTnjKNdkurDNvd3FHePHgTkPEUvjXYvg3brJj",
+        "5DiMVAp8WmFyWAwaTwAr7sU4K3brXcgNCBDbHoBWj3M46PiP",
+        "5EhREncHsntgJaax9YQphk1xN3LxPu2Rzbz4A3g7Ut8cRXWq",
+        "5C84NU2477uHCUF1A8rHb89sP2D2ZsnquPaGa2Htv75FN9gm",
+        "5HdZjrmNAkWQhYQUPNv7YRYnT4vyQswjbNm8eXBvULNQz5wH",
+        "5HjFLKpiCStQgRm6ZM1fT1R5pLKAqQdUG3uh7pvzaQfhdFuB",
+        "5ECktCamcAtBFJirEzvvJmXFxgLMCTAejhqZwLT1Dxn2fwB1",
+        "5EZbvFqx3j7ejqBSPWseif8xL3PwoqMQHdMT8rs9qWoHcdR3"
+    ]
+
+async function takeSnapshot(api, snapshotBalancesFile) {
+    const balances = {};
+    const accounts = await api.query.system.account.keys();
+
+    for (const key of accounts) {
+        const account = key.args[0].toString();
+        if (DISCARDED_ACCOUNTS.includes(account)) continue;
+        const { data: { free: balance } } = await api.query.system.account(account);
+        balances[account] = balance.toString();
+    }
+    fs.writeFileSync(snapshotBalancesFile, JSON.stringify(balances, null, 2));
+
+    console.log('Snapshot saved.');
+}
+
+// Restore balances from the snapshot
+async function restoreBalances(api, keyring, snapshotBalancesFile, custodySeed) {
+    const balances = JSON.parse(fs.readFileSync(snapshotBalancesFile));
+    const custodyAccount = keyring.addFromUri(custodySeed);
+    let nonce = (await api.query.system.account(custodyAccount.address)).nonce.toNumber();
+
+    for (const [account, balance] of Object.entries(balances)) {
+        const transfer = api.tx.balances.transferAllowDeath(account, balance);
+        try {
+            const hash = await transfer.signAndSend(custodyAccount, { nonce });
+            console.log(`Transferred to ${account}, transaction hash: ${hash.toHex()}`);
+            nonce++;
+            // Add a small delay between transactions to avoid priority issues
+            await new Promise(resolve => setTimeout(resolve, 500));
```

Comment:
This is interesting; what kind of priority issues are you referring to? Is it related to the ordering of transactions?

I'm asking because, in general, this kind of timeouts tend to be very fragile in terms of functionality and sub-optimal in terms of performance. It would be great to find a formal solution to the issue you are mitigating with the timeout.

Reply:
@drgora 
Actually my bad, it's just I had a "priority issue" but that was actually fixed by bumping the nonce, I did not need this timeout on top of that.

That being said, isn't there going to be any issue about submitting too many transactions within a same block? Or do they just sit in the mempool for next block(s)? 

I will remove the timeout for now

Reply:
Ok great, I'm glad to see this solved.

Also, you are right; even if you send more transactions than can fit in a single block, Substrate will handle everything as you would expect: they will stay in the mempool until they find a spot in a block,